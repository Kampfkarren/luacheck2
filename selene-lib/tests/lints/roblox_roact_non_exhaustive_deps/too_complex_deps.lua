local React

local function Component()
    local a = {}
    React.useEffect(function()
        print(a[b])
    end, { a[b] })
end

local function Component()
    local a = {}
    React.useEffect(function()
        print(a.b["c"]())
    end, { a.b["c"]() })
end

local function Component()
    local a = {}
    React.useEffect(function()
        print(a())
        -- FIXME: false negatives for function calls without indexing
    end, { a() })
end

local function Component()
    local a = {}
    React.useEffect(function()
        -- A function call in place of array brackets should NOT warn about complex expr
        -- due to lua-specific pattern of helper functions to patch holes in arrays
    end, a.b())

    React.useEffect(function()
        -- Now this should warn
    end, a.b(a.b()))

    React.useEffect(function()
        -- Now this should warn
    end, { a.b() })
end
