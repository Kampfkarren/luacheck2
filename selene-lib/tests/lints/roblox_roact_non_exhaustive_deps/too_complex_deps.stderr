error[roblox_roact_non_exhaustive_deps]: react hook useEffect has a complex expression in the dependency array
  ┌─ too_complex_deps.lua:7:10
  │
7 │     end, { a[b] })
  │          ^^^^^^^^
  │
  = help: extract it to a separate variable so it can be statically checked

error[roblox_roact_non_exhaustive_deps]: react hook useEffect has a complex expression in the dependency array
   ┌─ too_complex_deps.lua:14:10
   │
14 │     end, { a.b["c"]() })
   │          ^^^^^^^^^^^^^^
   │
   = help: extract it to a separate variable so it can be statically checked

error[roblox_roact_non_exhaustive_deps]: react hook useEffect has a complex expression in the dependency array
   ┌─ too_complex_deps.lua:34:10
   │
34 │     end, a.b(a.b()))
   │          ^^^^^^^^^^
   │
   = help: extract it to a separate variable so it can be statically checked

error[roblox_roact_non_exhaustive_deps]: react hook useEffect has a complex expression in the dependency array
   ┌─ too_complex_deps.lua:38:10
   │
38 │     end, { a.b() })
   │          ^^^^^^^^^
   │
   = help: extract it to a separate variable so it can be statically checked

